exist-core/src/main/java/org/exist/xmldb/RemoteCollection.java

```diff
@@ -110,37 +110,47 @@ public static RemoteCollection instance(final Leasable<XmlRpcClient> leasableXml
         }
     }
 
-    private RemoteCollection(final Leasable<XmlRpcClient> leasableXmlRpcClient, final Leasable<XmlRpcClient>.Lease  xmlRpcClientLease, final RemoteCollection parent, final XmldbURI path) throws XMLDBException {
+    private RemoteCollection(final Leasable<XmlRpcClient> leasableXmlRpcClient, final Leasable<XmlRpcClient>.Lease xmlRpcClientLease, final RemoteCollection parent, final XmldbURI path) throws XMLDBException {
         super(parent);
         this.path = path.toCollectionPathURI();
         this.leasableXmlRpcClient = leasableXmlRpcClient;
         this.xmlRpcClientLease = xmlRpcClientLease;
     }
 
+    protected final Object execute(String pMethodName, List pParams) throws XMLDBException {
+        if (xmlRpcClientLease.isClosed()) {
+            throw new XMLDBException(ErrorCodes.COLLECTION_CLOSED);
+        }
+        try {
+            return xmlRpcClientLease.get().execute(pMethodName, pParams);
+        } catch (final XmlRpcException xre) {
+            throw new XMLDBException(ErrorCodes.VENDOR_ERROR, xre.getMessage(), xre);
+        }
+    }
+
     @Override
     public void close() {
+        if (xmlRpcClientLease.isClosed()) {
```

Comment:
I don't think this is required as calling closed is `idempotent`

Reply:
No, because subsequent call on xmlRpcClientLease.close() will throw an error, that I don't want to have here. The `close()` method should gracefully ignore subsequent calls here.